lib/fpm/package/zip.rb

```diff
@@ -36,28 +36,12 @@ def input(input_path)
     dir.cleanup_build
   end # def input
 
-  # Output a tarball.
-  #
-  # If the output path ends predictably (like in .tar.gz) it will try to obey
-  # the compression type.
+  # Output a zipfile.
   def output(output_path)
     output_check(output_path)
-
-    files = Find.find(staging_path).to_a
-    safesystem("zip", output_path, *files)
-  end # def output
-
-  # Generate the proper tar flags based on the path name.
-  def tar_compression_flag(path)
-    case path
-      when /\.tar\.bz2$/
-        return "-j"
-      when /\.tar\.gz$|\.tgz$/
-        return "-z"
-      when /\.tar\.xz$/
-        return "-J"
-      else
-        return nil
+    realpath = Pathname.new(output_path).realdirpath.to_s
+    ::Dir.chdir(staging_path) do
+      safesystem("zip", "-9ry", realpath, ".")
```

Comment:
This is a behavioral change (adding -r and -y flags). What is the expected impact on users? I imagine the `-y` flag to build the zip will not work when extracting the resulting zip on Windows?